trace/trace_test.go

```diff
@@ -0,0 +1,125 @@
+package trace
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/mailgun/oxy/testutils"
+	"github.com/mailgun/oxy/utils"
+
+	. "gopkg.in/check.v1"
+)
+
+func TestTrace(t *testing.T) { TestingT(t) }
+
+type TraceSuite struct{}
+
+var _ = Suite(&TraceSuite{})
+
+func (s *TraceSuite) TestTraceSimple(c *C) {
+	handler := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("hello"))
```

Comment:
Oh, I know why - it's because the implicit WriteHeader is called after we return from the handler, let me add a logic to mirror it here to avoid confusion with 0's